web/packages/core/src/ruffle-player.tsx

```diff
@@ -966,6 +956,36 @@ export class RufflePlayer extends HTMLElement {
         }
     }
 
+    /**
+     * The effective config loaded upon `load`, possibly changing if window.RufflePlayer.config or this.config is changed
+     *
+     * @returns The effective config right now.
+     */
+    get loadedConfig(): URLLoadOptions | DataLoadOptions | undefined {
+        if (this.options) {
+            return {
+                ...DEFAULT_CONFIG,
+                // The default allowScriptAccess value for polyfilled elements is samedomain.
+                ...(this.isPolyfillElement && "url" in this.options
+                    ? {
+                        allowScriptAccess: parseAllowScriptAccess(
+                            "samedomain",
+                            this.options.url,
+                        )!,
+                    }
+                : {}),
+                ...(window.RufflePlayer?.config ?? {}),
+                // This will make extension config take precedence, if I move this one line higher self-hosted config will take precedence
+                // Let the heated discussions begin
```

Comment:
Comment is here to facilitate discussion. Start discussing below. This has come up before. Site owners have come out against it, mainly arguing against showSwfDownload being able to be changed by end-users. However, it is true that `window.open(document.querySelector("ruffle-player, ruffle-embed, ruffle-object").swfUrl, "_self")` can already be used, making this just a convenience method.